frontend/app/components/context-menus/column-context-menu/column-context-menu.controller.ts

```diff
@@ -0,0 +1,147 @@
+// -- copyright
+// OpenProject is a project management system.
+// Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License version 3.
+//
+// OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+// Copyright (C) 2006-2013 Jean-Philippe Lang
+// Copyright (C) 2010-2013 the ChiliProject Team
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+//
+// See doc/COPYRIGHT.rdoc for more details.
+// ++
+
+import {WorkPackageTableColumnsService} from '../../wp-fast-table/state/wp-table-columns.service';
+import {WorkPackageTableMetadataService} from '../../wp-fast-table/state/wp-table-metadata.service';
+
+angular
+  .module('openproject.workPackages')
+  .controller('ColumnContextMenuController', ColumnContextMenuController);
+
+function ColumnContextMenuController($scope,
+                                     columnContextMenu,
+                                     QueryService,
+                                     wpTableColumns:WorkPackageTableColumnsService,
+                                     wpTableMetadata:WorkPackageTableMetadataService,
+                                     I18n,
+                                     columnsModal) {
+
+  $scope.I18n = I18n;
+
+  $scope.$watch('column', function () {
```

Comment:
That's due to the query service and table columns still watching on `scope.columns`. I agree that we should move that to the states, but that involves quite a lot more than the table and also reaches into frontend work on the query following Jens' API redesign.

-> Fully agreed, but let's address this in a separate PR.